src/Spatial.Tests/Euclidean/Circle2DTests.cs

```diff
@@ -58,5 +58,34 @@ public void CircleFromThreePointsArgumentException()
 
             Assert.Throws<ArgumentException>(() => { Circle2D.FromPoints(p1, p2, p3); });
         }
+
+        [TestCase("0,0", 1.41421356 /*=sqrt(2)*/, "-1,-1", "+1,+1", "+1,+1", "-1,-1")]
```

Comment:
I would add some more tests, for which the start/end points of the line do not lie on the circle, just to make sure that we are not running on any edge case.

In addition there are test cases missing for no intersections as well as identical intersections (i.e. line tangent to the circle)

Reply:
Thanks for the comment about test cases.
This comment looks very similar to the bellow one, can I unify these 2 comments?

Reply:
You are right that they are very similar, however the line segment case requires further testing cases, since the intersection points (with the underlying line) might lie outside of the segment itself.

Reply:
done in the commit https://github.com/f-frhs/mathnet-spatial/commit/3d3748bca4569c6d5a7aed16c1c3957a1da9e8ff.